spring-cloud-gcp-autoconfigure/src/test/java/com/google/cloud/spring/autoconfigure/bigquery/GcpBigQueryAutoConfigurationTests.java

```diff
@@ -60,6 +61,107 @@ void testSettingBigQueryOptions() {
         });
   }
 
+  @Test
+  void testBigQuery_universeDomain() {
+    this.contextRunner
+        .withPropertyValues("spring.cloud.gcp.bigquery.universe-domain=myUniverseDomain")
+        .run(
+            ctx -> {
+              BigQueryOptions options = ctx.getBean(BigQuery.class).getOptions();
+              assertThat(options.getUniverseDomain()).isEqualTo("myUniverseDomain");
+              assertThat(options.getResolvedApiaryHost("bigquery"))
+                  .isEqualTo("https://bigquery.myUniverseDomain/");
+            });
+  }
+
+  @Test
+  void testBigQuery_noUniverseDomainAndEndpointSet_useClientDefault() {
+    this.contextRunner.run(
+        ctx -> {
+          BigQueryOptions options = ctx.getBean(BigQuery.class).getOptions();
+          assertThat(options.getUniverseDomain()).isNull();
+          assertThat(options.getResolvedApiaryHost("bigquery"))
+              .isEqualTo("https://bigquery.googleapis.com/");
+        });
+  }
+
+  @Test
+  void testBigQuery_endpoint() {
+    this.contextRunner
+        .withPropertyValues("spring.cloud.gcp.bigquery.endpoint=bigquery.example.com:443")
+        .run(
+            ctx -> {
+              BigQueryOptions options = ctx.getBean(BigQuery.class).getOptions();
+              assertThat(options.getResolvedApiaryHost("bigquery"))
+                  .isEqualTo("https://bigquery.example.com");
+            });
+  }
+
+  @Test
+  void testBigQuery_bothEndpointAndUniverseDomainSet() {
+    this.contextRunner
+        .withPropertyValues("spring.cloud.gcp.bigquery.endpoint=bigquery.example.com:123")
+        .withPropertyValues("spring.cloud.gcp.bigquery.universe-domain=myUniverseDomain")
+        .run(
+            ctx -> {
+              BigQueryOptions options = ctx.getBean(BigQuery.class).getOptions();
+              assertThat(options.getResolvedApiaryHost("bigquery"))
+                  .isEqualTo("https://bigquery.example.com");
+              assertThat(options.getUniverseDomain()).isEqualTo("myUniverseDomain");
+            });
+  }
+
+  @Test
+  void testBigQueryWrite_universeDomain() {
+    this.contextRunner
+        .withPropertyValues("spring.cloud.gcp.bigquery.universe-domain=myUniverseDomain")
+        .run(
+            ctx -> {
+              BigQueryWriteClient writeClient = ctx.getBean(BigQueryWriteClient.class);
+              assertThat(writeClient.getSettings().getUniverseDomain())
+                  .isEqualTo("myUniverseDomain");
+            });
+  }
+
+  @Test
+  void testBigQueryWrite_endpoint() {
+    this.contextRunner
+        .withPropertyValues(
+            "spring.cloud.gcp.bigquery.endpoint=bigquerystorage.example.com:123")
+        .run(
+            ctx -> {
+              BigQueryWriteClient client = ctx.getBean(BigQueryWriteClient.class);
+              assertThat(client.getSettings().getEndpoint())
+                  .isEqualTo("bigquerystorage.example.com:123");
```

Comment:
Please verify that it's intended to have getEndpoint not return the URI scheme, but getResolvedApiaryHost does return the URI scheme.

LGTM

Reply:
Yes, returning endpoint with this pattern is intended for Bigquerystorage which will otherwise throw an exception if it doesn't follow a specific convention. For example, passing in `bigquerystorage.example.com` without the port will result in:
```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.google.cloud.bigquery.storage.v1.BigQueryWriteClient]: Factory method 'bigQueryWriteClient' threw exception with message: invalid endpoint, expecting "<host>:<port>"
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:178)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:644)
	... 31 more
Caused by: java.lang.IllegalArgumentException: invalid endpoint, expecting "<host>:<port>"
```
On the other hand bigquery accepts either accepts user-provided endpoint as-is if one is provided or builds the host in the `https://service.universeDomain` format  if only the universe domain is provided:
https://github.com/googleapis/sdk-platform-java/blob/1a988df22f7e3d15ce6b121bf26897c59ab468e4/java-core/google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java#L868 but [doesn't contain an explicit check to verify this schema](https://github.com/googleapis/java-bigquery/blob/bb2cd471fda5a2218580c41839c09edc5fa6b483/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/spi/v2/HttpBigQueryRpc.java#L110). And in this PR, we reformat the provided endpoint to follow the same pattern that getResolvedApiaryHost() defaults to.